src/System/IO/Streams/SequenceId.hs

```diff
@@ -49,26 +55,39 @@ sequenceIdInputStream initSeqId getSeqId seqIdFaultHandler inStream =
 -- Example:
 --
 -- @
--- (outStream', getSeqId) <- 'sequenceIdOutputStream' 1 outStream
--- return $ 'System.IO.Streams.Combinators.contramapM' (addSeqId getSeqId) outStream'
+-- ghci> import qualified System.IO.Streams as Streams
+-- ghci> (os, getList) <- Streams.listOutputStream :: IO (OutputStream (Int,Int), IO [(Int,Int)])
+-- ghci> (outStream', getSeqId) <- sequenceIdOutputStream 0 (\seqId a -> (seqId, a)) os
+-- ghci> Streams.write (Just 6) outStream'
+-- ghci> Streams.write (Just 7) outStream'
+-- ghci> getList
+-- [(1,6),(2,7)]
+-- ghci> getSeqId
+-- 2
+-- ghci> Streams.write (Just 6) outStream'
+-- ghci> Streams.write (Just 7) outStream'
+-- ghci> getList
+-- [(1,6),(2,7)]
 -- @
 sequenceIdOutputStream :: Integral s
-                       => s                   -- ^ Initial sequence ID
-                       -> (s -> a -> b)       -- ^ Transformation function
-                       -> OutputStream b      -- ^ 'System.IO.Streams.OutputStream' to count the elements of
-                       -> IO (OutputStream a) -- ^ ('IO' 'SequenceId') is the action to run to get the current sequence ID
+                       => s                         -- ^ Initial sequence ID
+                       -> (s -> a -> b)             -- ^ Transformation function
+                       -> OutputStream b            -- ^ 'System.IO.Streams.OutputStream' to count the elements of
+                       -> IO (OutputStream a, IO s) -- ^ returns a new stream as well as an IO action to fetch and reset the updated seed value
```

Comment:
Why not make it so that you can assert the sequence id state? I think merely resetting is not flexible enough for our purposes. For example, I think in a lot of places we reset the sequence stream to be 2 instead of 1 after login, but you'd want to be able to reset it to 1, for example.

Reply:
This is a really good point and I think it rules out using `inputFoldM` (and `outputFoldM`) as shipped by `io-streams`.

Reply:
yeah good catch. I just pushed up a new version that slightly modifies `{input,output}FoldM` so that we can assert the sequence id state. Let me know what you guys think @GambolingPangolin  @wraithm